step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -69,22 +75,54 @@ public ImportResult importArtefact(ExecutionContext context, Map<String, String>
         AutomationPackage automationPackage = getAutomationPackageForContext(repositoryParameters);
 
         String planId = repositoryParameters.get(RepositoryObjectReference.PLAN_ID);
+        ImportResult result = new ImportResult();
+        result.setPlanId(planId);
 
-        Plan plan = automationPackageManager.getPackagePlans(automationPackage.getId().toString())
+        Plan plan = automationPackageManager.getPackagePlans(automationPackage.getId())
                 .stream()
                 .filter(p -> p.getId().toString().equals(planId)).findFirst().orElse(null);
         if (plan == null) {
-            return null;
+            // failed result
+            result.setErrors(List.of("Automation package " + automationPackage.getAttribute(AbstractOrganizableObject.NAME) + " has no plan with id=" + planId));
+            return result;
         }
-
         enrichPlan(context, plan);
-        context.getPlanAccessor().save(plan);
 
-        ImportResult result = new ImportResult();
-        result.setPlanId(plan.getId().toString());
+        PlanAccessor planAccessor = context.getPlanAccessor();
+        if (!(planAccessor instanceof LayeredAccessor)) {
+            result.setErrors(List.of(planAccessor.getClass() + " is not layered"));
+            return result;
+        }
+
+        planAccessor.save(plan);
 
-        plan.getFunctions().iterator().forEachRemaining(f -> (context.get(FunctionAccessor.class)).save(f));
-        automationPackageManager.getPackageFunctions(automationPackage.getId().toString()).forEach(f -> context.get(FunctionAccessor.class).save(f));
+        FunctionAccessor functionAccessor = context.get(FunctionAccessor.class);
+        if(!(functionAccessor instanceof LayeredAccessor)){
```

Comment:
This check makes sense. Please introduce a generic method to check this to avoid duplicated code